components/WeekdayPicker.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
+import { useThemeColor } from '../components/Themed'
+interface Props {
+  weekdays: number[]
+  setWeekdays: React.Dispatch<React.SetStateAction<number[]>>
+}
+export default function DayPicker({ weekdays, setWeekdays }: Props) {
+  function daysIO(v: number) {
+    if (weekdays.includes(v)) {
+      const weekdayRemoved = weekdays.filter((element) => element !== v)
+      setWeekdays(weekdayRemoved)
+    } else {
+      setWeekdays([...weekdays, v])
+    }
+  }
+
+  const primaryColor = useThemeColor({}, 'primary')
+
+  const styles = StyleSheet.create({
+    box: {
+      width: 40,
+      height: 40,
+      borderRadius: 20,
+      justifyContent: 'center',
+      alignItems: 'center',
+      color: 'white',
+      marginVertical: 10,
+    },
+    box1: {
```

Comment:
Repeating this logic can be avoided by extracting the styles outside of the component and using a ternary inside the render function to change the styles